src/display/text_layer.js

```diff
@@ -595,6 +597,7 @@ class TextLayerRenderTask {
     this._capability = createPromiseCapability();
     this._renderTimer = null;
     this._bounds = [];
+    this._devicePixelRatio = window.devicePixelRatio;
```

Comment:
To prevent issues in non-browser environments, please change this to:
```suggestion
    this._devicePixelRatio = globalThis.devicePixelRatio || 1;
```